test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,1047 @@
+    RUN_AFTER(c.hmiNotification, pWait)
+  end)
+  :Do(function(_, data)
+      c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", c.getHMIResponseParams(consentRPC, true))
+      EXPECT_HMICALL(c.getHMIEventName(pRPC), c.getHMIRequestParams(pRPC, pModuleType, pAppId))
+      :Do(function(_, data2)
+          c.getHMIConnection():SendResponse(data2.id, data2.method, "SUCCESS", c.getHMIResponseParams(pRPC, pModuleType))
+        end)
+    end)
+  c.getMobileSession(pAppId):ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+function c.rpcAllowedWithConsentError(pModuleType, pAppId, pRPC, pRequestID, pMethodName, pResetPeriod, pWait)
+  if not pAppId then pAppId = 1 end
+  local cid = c.getMobileSession(pAppId):SendRPC(c.getAppEventName(pRPC), c.getAppRequestParams(pRPC, pModuleType))
+  local consentRPC = "GetInteriorVehicleDataConsent"
+  EXPECT_HMICALL(c.getHMIEventName(consentRPC), c.getHMIRequestParams(consentRPC, pModuleType, pAppId))
+  :Do(function()
+    c.hmiNotification(pRequestID, pMethodName, pResetPeriod)
+    RUN_AFTER(c.hmiNotification, pWait)
+  end)
+  :Do(function(_, _)
+    -- HMI does not respond
+    end)
+  c.getMobileSession(pAppId):ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR" })
+end
+
+function c.backupHMICapabilities()
+  local hmiCapabilitiesFile = commonFunctions:read_parameter_from_smart_device_link_ini("HMICapabilities")
+  commonPreconditions:BackupFile(hmiCapabilitiesFile)
+end
+
+local function audibleState(pAppId)
+  if not pAppId then pAppId = 1 end
+  local appParams = config["application" .. pAppId].registerAppInterfaceParams
+  local audibleStateValue
+  if appParams.isMediaApplication == true then
+    audibleStateValue = "AUDIBLE"
+  else
+    audibleStateValue = "NOT_AUDIBLE"
+  end
+  return audibleStateValue
+end
+
+function c.activateApp(pAppId)
+  if not pAppId then pAppId = 1 end
+  local pHMIAppId = c.getHMIAppId(pAppId)
+  local mobSession = c.getMobileSession(pAppId)
+  local requestId = test.hmiConnection:SendRequest("SDL.ActivateApp", { appID = pHMIAppId })
+  EXPECT_HMIRESPONSE(requestId)
+  mobSession:ExpectNotification("OnHMIStatus", { hmiLevel = "FULL", audioStreamingState = audibleState(pAppId),
+      systemContext = "MAIN" })
+  utils.wait()
+end
+
+function c.hmiNotification(pRequestID, pMethodName, pResetPeriod)
+  if not pRequestID then pRequestID = c.getHMIAppId() end
+  if not pResetPeriod then pResetPeriod = 3000 end
+  c.getHMIConnection():SendNotification("BasicCommunication.OnResetTimeout",
+    { requestID = pRequestID,
+    methodName = pMethodName,
+    resetPeriod = pResetPeriod
+  })
+end
+
+function c.sendLocation( pRequestID, pMethodName, pResetPeriod, pWait )
+  local cid = c.getMobileSession():SendRPC("SendLocation", sendLocationRequestParams)
+  EXPECT_HMICALL("Navigation.SendLocation", { appID = c.getHMIAppId() })
+  :Do(function()
+    c.hmiNotification(pRequestID, pMethodName, pResetPeriod)
```

Comment:
RUN_AFTER is not work in such implementation.
Use parameters for notification from received request, not as main function parameters

The solution is
```
EXPECT_HMICALL("Navigation.SendLocation", { appID = c.getHMIAppId() })
  :Do(function(_, data)
     c.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
     local function onResetTimeoutCall()
        c.hmiNotification(data.id, data.method, pResetPeriod)
     end
    RUN_AFTER(onResetTimeoutCall, pWait)
  end)


   local function sendResponse()
      c.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
    end
  RUN_AFTER(sendResponse, pTimeToResponse)
```
Also in case there is the need to not send response please configure the function to send and not send response.